src/rules/no-useless-path-segments.js

```diff
@@ -45,65 +62,99 @@ module.exports = {
         type: 'object',
         properties: {
           commonjs: { type: 'boolean' },
+          detectUselessIndex: { type: 'boolean' },
         },
         additionalProperties: false,
       },
     ],
 
     fixable: 'code',
+    messages: {
+      uselessPath: 'Useless path segments for "{{ path }}", should be "{{ proposedPath }}"',
+    },
   },
 
-  create: function (context) {
+  create(context) {
     const currentDir = path.dirname(context.getFilename())
+    const options = context.options[0]
 
     function checkSourceValue(source) {
-      const { value } = source
+      const { value: importPath } = source
 
-      function report(proposed) {
+      function report(proposedPath) {
         context.report({
           node: source,
-          message: `Useless path segments for "${value}", should be "${proposed}"`,
-          fix: fixer => fixer.replaceText(source, JSON.stringify(proposed)),
+          messageId: 'uselessPath',
+          data: {
+            path: importPath,
+            proposedPath,
+          },
+          fix: fixer => proposedPath && fixer.replaceText(source, JSON.stringify(proposedPath)),
```

Comment:
i think that when proposedPath is not present, `fix` shouldn’t be either

Reply:
I've updated the implementation for this case and added a better output handling:
- `fix` is only set if there is a possible fix
- If the auto-fix is not possible because of an ambiguous path, a dedicated message which contains the colliding paths is shown